annotation/src/main/java/io/smallrye/common/annotation/Experimental.java

```diff
@@ -0,0 +1,18 @@
+package io.smallrye.common.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that specifies that an element is experimental and may change without notice.
+ */
+@Inherited
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
+public @interface Experimental {
+    String value() default "";
```

Comment:
I guess the value is meant to be some explanation as to why it is experimental?

Reply:
Also I would add `@Documented` to this.

Reply:
Yes, let me add a javadoc in there.